### PR TITLE
Adding edited emails.

### PR DESCRIPTION
**Pull request by Anish-Agnihotri**
##### Description

This PR introduces one email template (update_notification). It is fully completed, and 100% mobile responsive. In a future PR, I will be adding its routing into Django.

*new:* In addition, this also fixes common issues across all email files, and resolves any pre-existing merge errors. 

##### Checklist

- [x] Linter status: 100% pass (failing checks are because of other files.). 
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)
- [x] Checklist for #2289 has been met

##### Affected core subsystem(s)

No core systems have been affected until the email is implemented into the respective `emails.py` and `urls.py` files. I will be completing this integration as well after this PR by using dummy variables.

##### Testing
This change has been tested across two operating systems using multiple integration tests. In order to test for yourself, you have two options: 

1. Since the email has not been implemented into `urls.py`, you can replace an existing email (I personally used `new_bounty.html`) with the content of the pushed email to see data.
2. You can setup the routing yourself and see the magic happen. 

##### Refers/Fixes

Resolves the following issues: 

Update Notification, #2289, #2088

### CODE REVIEW

File: app/retail/templates/emails/grants/support_cancellation.html [hunk 1]
```diff
@@ -60,126 +64,163 @@
...
-                <h4><strong>{{current_goal_amount}}</strong></h4> <!-- Input goal funding value -->
-                <p>{%trans "Goal" %}</p>
-            </div>
-        </div>
+  <a href="https://github.com/{{ github_username }}">
+    <img src="{{ github_username }}" />
+  </a>
+  <!-- width="50" height="50"Input project image here -->
+  <br>
+  <a class="grantInfoName" href="https://github.com/{{ github_username }}">{{ grant_name }}</a>
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/support_cancellation.html]: Shouldn't we link to the grant details page here?

File: app/retail/templates/emails/grants/support_cancellation.html [hunk 2]
```diff
@@ -1,56 +1,60 @@
+{% extends '../emails/template.html' %} 
+{% comment %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/support_cancellation.html]: Trailing whitespace.

File: app/retail/templates/emails/grants/subscription_terminated.txt [hunk 3]
```diff
@@ -1,21 +1,21 @@
 {% load i18n humanize %}
 
-{% trans "Grant" %} {{grant_name}} {% trans "has been cancelled" %}
+{% trans "Grant" %} {{ grant_name }} {% trans "has been cancelled" %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.txt]: All of these variables need updated to take the subscription object and use it's associated attributes.

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 4]
```diff
@@ -60,93 +64,117 @@
...
-<p>{%trans "You can see the subscription cancelation transaction" %} <a href="https://etherscan.com/{{etherscan_link}}">{%trans "here." %}</a></p> <!-- Input cancelation link here -->
+<p>{%trans "You can see the subscription cancelation transaction" %}
+  <a href="https://etherscan.com/{{ etherscan_link }}">{%trans "here." %}</a>
+</p>
+<!-- Input cancelation link here -->
 <br>
-<a class="updatesButton" href="https://etherscan.com/{{etherscan_link}}">{%trans "View Transaction" %}</a> <!-- Input link to view transaction button here --> 
-<br>
-
+<a class="updatesButton" href="https://etherscan.com/{{ etherscan_link }}">{%trans "View Transaction" %}</a>
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: All of the `trans` blocks need fixed for whitespace...

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 5]
```diff
@@ -60,93 +64,117 @@
...
+    <img src="{{ github_username }}" />
+  </a>
+  <!-- width="50" height="50"Input project image here -->
+  <br>
+  <a class="grantInfoName" href="https://github.com/{{ github_username }}">{{ github_username }}</a>
+  <!-- Input project name here -->
+  <p>{{ grant_description }}</p>
+  <!-- Input project description here -->
+  <div class="progress">
+    <h3 style="text-transform: none;">{{ funding_status }}</h3>
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: All of the template vars throughout this file need to reference the `Grant` object and it's associated attributes.

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 6]
```diff
@@ -60,93 +64,117 @@
...
-                <h4><strong>{{goal_funding}}</strong></h4> <!-- Input goal funding value -->
-                <p>{%trans "Goal" %}</p>
-            </div>
-        </div>
+  <a href="https://github.com/{{ github_username }}">
+    <img src="{{ github_username }}" />
+  </a>
+  <!-- width="50" height="50"Input project image here -->
+  <br>
+  <a class="grantInfoName" href="https://github.com/{{ github_username }}">{{ github_username }}</a>
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: This should be linking back to the grant page vs the user's github page.

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 7]
```diff
@@ -60,93 +64,117 @@
...
     height: 45px;
     background-color: #979797;
     float: left;
     transform: translateY(-20%);
     margin-right: 20px;
-}
-.funding > p {
+  }
+
+  .funding>p {
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: ?

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 8]
```diff
@@ -1,56 +1,60 @@
...
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU Affero General Public License for more details.
 
+ You should have received a copy of the GNU Affero General Public License
+ along with this program. If not, see <http://www.gnu.org/licenses/>. 
+ 
 {% endcomment %}
-{% load i18n humanize %}
-{% load i18n static %}
+{% load i18n humanize static %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: No need for humanize in here.

File: app/retail/templates/emails/grants/subscription_terminated.html [hunk 9]
```diff
@@ -1,56 +1,60 @@
+{% extends '../emails/template.html' %} 
+{% comment %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/subscription_terminated.html]: whitespace

File: app/retail/templates/emails/grants/new_supporter.txt [hunk 10]
```diff
@@ -2,10 +2,9 @@
...
 {% trans "You have a new supporter!" %}
 
-{{github_username}}
+{{ github_username }}
 {%trans "User' has begun contributing" %}
-{{variable_amount_contribution}}
+{{ variable_amount_contribution }}
 
-{%trans "You can see a transaction history and withdraw funds" %} {{transaction_history}} {%trans "here." %}
+{%trans "You can see transaction history" %} {{ transaction_history }} {%trans "here." %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/new_supporter.txt]: trans whitespace needs fixed throughout

File: app/retail/templates/emails/grants/new_supporter.txt [hunk 11]
```diff
@@ -2,10 +2,9 @@
 
 {% trans "You have a new supporter!" %}
 
-{{github_username}}
+{{ github_username }}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/new_supporter.txt]: All of these vars should reference the input `Grant` object...

File: app/retail/templates/emails/grants/new_supporter.html [hunk 12]
```diff
@@ -1,94 +1,90 @@
...
 <br>
-<p>{%trans "You can see a transaction history and withdraw funds" %} <a href="https://gitcoin.co/{{transaction_history}}">{%trans "here." %}</a></p> <!-- Input link to transaction history here -->
-
-<br>
-<a class="withdrawButton" href="https://gitcoin.co/{{withdraw_link}}">{%trans "Withdraw" %}</a> <!-- Input link to withdraw button here --> 
-
-<br>
+<!-- Input link to transaction history here -->
+<p>{%trans "You can see transaction history" %}
+  <a href="https://gitcoin.co/{{ transaction_history }}">{%trans "here." %}</a>
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/new_supporter.html]: This needs to use `{% url %}` references vs hardcoded paths.  - Throughout all of these files.

File: app/retail/templates/emails/grants/new_supporter.html [hunk 13]
```diff
@@ -1,94 +1,90 @@
-{% extends 'emails/template.html' %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/new_supporter.html]: Unnecessary trailing whitespace, vars need to use the input `subscription` and/or `grant` objects.  None of these variables will exist.

File: app/retail/templates/emails/grants/new_supporter.html [hunk 14]
```diff
@@ -1,94 +1,90 @@
...
-    along with this program. If not, see <http://www.gnu.org/licenses/>.
-
-{% endcomment %}
-{% load i18n humanize %}
-{% load i18n static %}
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+  
+{% endcomment %} 
+{% load i18n humanize static %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/new_supporter.html]: No need for humanize here

File: app/retail/templates/emails/grants/grant_cancellation.html [hunk 15]
```diff
@@ -1,46 +1,44 @@
-{% extends 'emails/template.html' %}
```
Comment (mbeacom) [inline app/retail/templates/emails/grants/grant_cancellation.html]: No reason for the trailing whitespace additions here.
All vars need updated to use the input grant object, none of these variables will exist.
Follow django proper whitespace for template variables and functions.
URLs need updated to use the `url` reverse.
We need all of these templates hooked up to the respective email func and passing the grant and/or subscription objects into the email methods on event.

**Comment by Anish-Agnihotri** [comment]
Thanks for review @mbeacom. Always appreciated. I'll refactor and resubmit within 24hr.

**Comment by mbeacom** [comment]
@Anish-Agnihotri Is there anything I can do to help you with this?  Are you still interested in moving this forward?

**Comment by Anish-Agnihotri** [comment]
@mbeacom For sure! Been done with a fever so have been unable to refactor. Thanks a bunch for the review, I'll be pushing out a new commit within the next day.

**Comment by Anish-Agnihotri** [comment]
Just fixed the other email. This PR should be updated within the next day, if not the next few hours. Just figuring out some small things in `urls.py` and making sure I move over the objects to their appropriate `grant` and `subscription` objects.

Other than that, just basic aesthetic changes.

**Comment by Anish-Agnihotri** [comment]
Status check on commit (updating as I go): 

Grant Cancellation Email: 👍 
1. Removed trailing white-spaces.
2. Updated to use grants object `{{ grants.title }}`.
3. Updated Django tag whitespacing. 
4. Removed unnecessary humanize load since no numbers were being parsed. 

Grants Email: 
New Supporter Email: 
Subscription Terminated Email:
Support Cancellation Email: 
Thank You For Supporting Email:
Update Notification Email:

**Comment by Anish-Agnihotri** [comment]
Ready for re-review @mbeacom @captnseagraves.

Thanks!

**Pull request merged**